shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/network/core/NetworkClient.kt

```diff
@@ -25,19 +26,9 @@ class NetworkClient {
 
     constructor(engine: HttpClientEngine? = null) {
         if (engine == null) {
-            client = HttpClient() {
-                install(Logging)
-                install(ContentNegotiation) {
-                    json(json)
-                }
-            }
+            client = HttpClient(clientConfig())
         } else {
-            client = HttpClient(engine) {
-                install(Logging)
-                install(ContentNegotiation) {
-                    json(json)
-                }
-            }
+            client = HttpClient(engine, clientConfig())
         }
```

Comment:
Can we also add parameter names here for the `HttpClient`? and we should lift the `client` assignment out of the `if` block as per the Android convention?

```
        client = if (engine != null) {
            HttpClient(engine = engine, block = clientConfig())
        } else {
            HttpClient(block = clientConfig())
        }
```